step_command.go

```diff
@@ -32,9 +32,7 @@ type CommandStep struct {
 	Signature *Signature        `yaml:"signature,omitempty"`
 	Matrix    *Matrix           `yaml:"matrix,omitempty"`
 
-	// RemainingFields stores any other top-level mapping items so they at least
-	// survive an unmarshal-marshal round-trip.
-	RemainingFields map[string]any `yaml:",inline"`
+	BaseStep BaseStep `yaml:",inline"`
```

Comment:
can we embed BaseStep and have it be inline, for better usability? that way users defining pipelines in go could say:

```Go
CommandStep{
  Command: "echo 'hi there'",
  Plugins: &Plugins{/* ... */?,
  Key: "very-important-message",
}
```

it feels like making customers use a `BaseStep` to define their keys, labels etc when that concept doesn't exist in the pipeline feels like the abstraction leaking out into userspace.

Reply:
If it is embedded with `yaml:",inline"`, there will be an unmarshalling error because of multiple inline fields (given that `BaseStep` and `BaseStep.RemainingFields` are both accessible as "direct" fields of `CommandStep`). That could be fixed.

If it is embedded without `yaml:",inline"`, the current implementation of `inlineFriendlyMarshalJSON` marshals the contents of `BaseStep` under an empty string key. That could also be fixed.

Either way, while it means we can write `var cs CommandStep; cs.Key = ...`, it's not enough to avoid referring to `BaseStep` in a struct literal, e.g. https://go.dev/play/p/ebVGXHsYpgW